### PR TITLE
Add completion for values in comparisons when comparing Enums

### PR DESCRIPTION
**Pull request by MartinGC94**
# PR Summary
Adds completion for scenarios like `Get-VM | where State -eq <Tab>` and `Get-VM | where {$_.State -eq <Tab>}`
This only works when the left side of the comparison is an enum.  
Also, due to the bug mentioned here: https://github.com/PowerShell/PowerShell/issues/17653 it doesn't work when you explicitly specify the `Property` parameter like this: `Get-VM | Where-Object -Property State -EQ ` you can work around this by either not specifying `-Property` or by explicitly specifying `-Value`

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
Hmm, looking that last commit I think about Roslin _trivia_. :-)

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -3984,6 +3989,37 @@ private static void NativeCompletionMemberName(CompletionContext context, List<C
             result.Add(CompletionResult.Null);
         }
 
+        private static void NativeCompletionMemberValue(CompletionContext context, List<CompletionResult> result, CommandAst commandAst, string propertyName)
+        {
+            string wordToComplete = context.WordToComplete.Trim('"', '\'');
+            IEnumerable<PSTypeName> prevType = GetInferenceTypes(context, commandAst);
+            if (prevType is not null)
+            {
+                foreach (var type in prevType)
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:3998]:
```suggestion
            IEnumerable<PSTypeName> prevTypes = GetInferenceTypes(context, commandAst);
            if (prevTypes is not null)
            {
                foreach (var type in prevTypes)
```

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 2]
```diff
@@ -3984,6 +3989,37 @@ private static void NativeCompletionMemberName(CompletionContext context, List<C
...
+                    if (property is not null && property.PropertyType.IsEnum)
+                    {
+                        foreach (var value in property.PropertyType.GetEnumNames())
+                        {
+                            if (value.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase))
+                            {
+                                result.Add(new CompletionResult(value, value, CompletionResultType.ParameterValue, value));
+                            }
+                        }
+                        break;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:4015]:
```suggestion
                        }

                        break;
```

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 3]
```diff
@@ -5902,6 +5938,44 @@ internal static List<CompletionResult> CompleteMember(CompletionContext context,
...
+                if (type.IsEnum)
+                {
+                    foreach (var name in type.GetEnumNames())
+                    {
+                        if (name.StartsWith(context.WordToComplete, StringComparison.OrdinalIgnoreCase))
+                        {
+                            result.Add(new CompletionResult($"'{name}'", name, CompletionResultType.ParameterValue, name));
+                        }
+                    }
+                    break;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:5973]:
```suggestion
                    }

                    break;
```

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 4]
```diff
@@ -5902,6 +5938,44 @@ internal static List<CompletionResult> CompleteMember(CompletionContext context,
...
+                }
+            }
+            return result;
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:5976]:
```suggestion
            }

            return result;
```

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]
LGTM!

**Pull request merged**